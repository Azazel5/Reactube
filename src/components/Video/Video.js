import React from 'react'
import './Video.scss'

const BASE_EMBED_URL = 'https://www.youtube.com/embed/';

/**
 * There's a neat CSS trick used here to make sure this video componenet looks good in screens 
 * of all sizes. It forces the screen to have 16:9 ratio in all i.e. the ::before selector we 
 * use. We also set a padding-top to 9/16 (56.25%). The trick is to understand that all percentage
 * values in padding and margin always refer to the elementâ€™s width. Eg - 
 * width: 100px; padding-top: 10%; (the pddding will always equal 10px. If it increases, the padding
 * increases too). 
 */
const video = (props) => {
    if (!props.id) {
        return null
    }

    // Add ?&autoplay=1 to autoplay when everything's built
    const embedURL = `${BASE_EMBED_URL}${props.id}`
    console.log(embedURL)
    return (
        <div className="video-container">
            <div className="video">
                <iframe className="video-player" width={'560'} height={'315'} src={embedURL} 
                    frameBorder='0' allow='autoplay; encrypted-media' 
                    allowFullScreen title="unique_title"/>
            </div>
        </div>

    )
}

export default video