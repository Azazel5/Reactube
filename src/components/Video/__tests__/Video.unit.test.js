import React from 'react'
import {shallow} from 'enzyme'
import Video from '../Video'

describe('<Video />', () => {
    test('renders a <Video /> component with an id', () => {
        const wrapper = shallow(<Video id="some_id"/>)
        expect(wrapper).toMatchSnapshot()
    })

    test('renders an empty <Video />', () => {
        const wrapper = shallow(<Video />)
        expect(wrapper).toMatchSnapshot()
    })
})