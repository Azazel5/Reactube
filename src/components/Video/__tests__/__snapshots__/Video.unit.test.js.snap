// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video /> renders a <Video /> component with an id 1`] = `
<div
  className="video-container"
>
  <div
    className="video"
  >
    <iframe
      allow="autoplay; encrypted-media"
      allowFullScreen={true}
      className="video-player"
      frameBorder="0"
      height="315"
      src="https://www.youtube.com/embed/some_id"
      title="unique_title"
      width="560"
    />
  </div>
</div>
`;

exports[`<Video /> renders an empty <Video /> 1`] = `""`;
